### PR TITLE
fix: permissions for viewing the allocate worker form

### PR DESCRIPTION
**Pull request by chrishutchinson**
**What**  
When viewing the add or remove allocated worker form, it wasn't doing a full permission check. This updates that logic to repurpose the logic we use for showing / hiding the "Allocated worker" button. Yay for consistency

**Anything else?**

- ~Have you added any new third-party libraries?~
- ~Are any new environment variables or configuration values needed?~
- ~Anything else in this PR that isn't covered by the what/why?~

### CODE REVIEW

File: lib/permissions.ts [hunk 1]
```diff
@@ -60,3 +60,27 @@ export const canManageCases = (
...
+
+  if (user.hasChildrenPermissions && person.contextFlag === 'C') {
+    // Children's doesn't require `hasAllocationsPermissions`, as anyone can allocate in CFS
+    return true;
+  }
+
+  if (
+    user.hasAdultPermissions &&
+    person.contextFlag === 'A' &&
+    (user.hasAllocationsPermissions || false)
```
Comment (Renny4Real) [inline lib/permissions.ts:80]: Why do we do an OR operation here: `(user.hasAllocationsPermissions || false)`?

Wouldn't `user.hasAllocationsPermissions` be either `true` or `false`, and then when compared to `false`, the OR operation would return the value of `user.hasAllocationsPermissions`. Is this right? What does `|| false` help us to do or check for? 🤔

**Review by Renny4Real — APPROVED** [review]
🚀

File: lib/permissions.ts (hunk 1 above)
Comment (chrishutchinson) [reply lib/permissions.ts:80]: Entertainingly the type def for `user.hasAllocationPermissions` is:

```ts
  hasAllocationsPermissions?: boolean;
```

meaning the potential values here are `true`, `false`, and `undefined`.

The `|| false` I've added effectively defaults it to false if it isn't set.

With that in mind, I actually think the type definition is wrong. I've created this ticket – https://hackney.atlassian.net/browse/SCT-233 – to refactor this type across the app.

**Comment by chrishutchinson** [comment]
Closing this as it’s gotten a bit stale and doesn’t appear to have an associated ticket. I will review if we actually need it next sprint, and re-create a ticket if so.

**Pull request closed without merging**